src/task/task.service.ts

```diff
@@ -1,66 +1,100 @@
-import { Injectable } from "@nestjs/common";
+import { Injectable, UnauthorizedException } from "@nestjs/common";
 import { InjectRepository } from "@nestjs/typeorm";
 import { InsertResult, Repository } from "typeorm";
 
 import { TaskDTO } from "./task.dto";
 import { Task } from "./task.entity";
 import { TaskMapper } from "./task.mapper";
+import { UserDTO } from "src/user/user.dto";
+import { User } from "src/user/user.entity";
+import { UserMapper } from "src/user/user.mapper";
+import { UserService } from "src/user/user.service";
 
 @Injectable()
 export class TaskService {
 
 	public constructor(
 		@InjectRepository(Task)
 		private taskRepo: Repository<Task>,
-		private taskMapper: TaskMapper
+		private taskMapper: TaskMapper,
+		private userService: UserService,
+		private userMapper: UserMapper
 	) { }
 
-	public async addTask(taskDTO: TaskDTO): Promise<InsertResult> {
+	public async addTask(taskDTO: TaskDTO, userId: string): Promise<InsertResult> {
+		taskDTO.user = Number(userId);
 		const task: Task = await this.taskMapper.toEntity(taskDTO);
 		return this.taskRepo.insert(task);
 	}
 
-	public async getAllTasks(): Promise<TaskDTO[]> {
+	public async getAllTasksByUserId(userId: string): Promise<TaskDTO[]> {
+		const userDto: UserDTO = await this.userService.getUserById(Number(userId));
+		const user: User = await this.userMapper.toEntity(userDto);
 		const tasks: Task[] = await this.taskRepo.find({
+			where: { user: user },
 			relations: ["user"]
 		});
 		return this.taskMapper.toDTOs(tasks);
 	}
 
-	public async getTaskById(id: number): Promise<TaskDTO> {
-		//if you have the same names of properties, you can get rid of one of them for eg: { id: id } will be { id }
-		//return this.taskRepo.findOne({ where: { id: id } });
-		const task: Task = await this.taskRepo.findOne({
-			where: { id: id },
-			relations: ["user"]
-		});
-		return this.taskMapper.toDTO(task);
+	public async getTaskById(id: number, userId: string): Promise<TaskDTO> {
+		const userDto: UserDTO = await this.userService.getUserById(Number(userId));
+		const user: User = await this.userMapper.toEntity(userDto);
+		try {
+			const task: Task = await this.taskRepo.findOne({
+				where: {
+					id: id,
+					user: user
+				},
+				relations: ["user"]
+			});
+			return this.taskMapper.toDTO(task);
+		} catch {
+			throw new UnauthorizedException();
+		}
 	}
 
-	public async updateTaskById(taskDTO: TaskDTO): Promise<TaskDTO> {
-		const task: Task = await this.taskMapper.toEntity(taskDTO);
-		await this.taskRepo.update(taskDTO.id, task);
-		const taskUpdated: Task = await this.taskRepo.findOne({ where: { id: taskDTO.id } });
-		return this.taskMapper.toDTO(taskUpdated);
+	public async updateTaskById(taskDTO: TaskDTO, userId: string): Promise<TaskDTO> {
+		const userDto: UserDTO = await this.userService.getUserById(Number(userId));
+		const user: User = await this.userMapper.toEntity(userDto);
+		if(taskDTO.user === user.id) {
```

Comment:
```suggestion
		if (taskDTO.userId === user.id) {
```
we should rename the property to userId